pkg/virt-controller/watch/network.go

```diff
@@ -91,19 +93,51 @@ func trimDoneInterfaceRequests(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInst
 	vm.Status.InterfaceRequests = updateIfaceRequests
 }
 
-func handleDynamicInterfaceRequests(vm *v1.VirtualMachine) {
+func handleDynamicInterfaceRequests(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig) error {
 	if len(vm.Status.InterfaceRequests) == 0 {
-		return
+		return nil
+	}
+
+	// When a VMI is created, in some cases if the VM has no interfaces, the default interface is auto added to it.
+	// Since the auto added interface is defined only on the VMI spec, explicitly add it to the VM before hot-plugging another interface.
+	err := addDefaultInterfaceToVm(vm, vmi, clusterConfig)
+	if err != nil {
+		return err
 	}
 
 	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
 	for i := range vm.Status.InterfaceRequests {
-		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
+		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMTemplateSpec(
 			vmTemplateCopy, &vm.Status.InterfaceRequests[i])
 	}
 	vm.Spec.Template.Spec = *vmTemplateCopy
 
-	return
+	return nil
+}
+
+func addDefaultInterfaceToVm(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig) error {
+	vmTemplateCopy := vm.Spec.Template.Spec.DeepCopy()
+	if vmispec.LookupPodNetwork(vmi.Spec.Networks) != nil && len(vmTemplateCopy.Domain.Devices.Interfaces) == 0 {
+		if hasAddRequest(vm) {
+			err := clusterConfig.SetVMISpecDefaultNetworkInterface(vmTemplateCopy)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	vm.Spec.Template.Spec = *vmTemplateCopy
```

Comment:
Maybe it would be safer not to mutate it here but just return the copy.
Then, that copy can be used to process `ApplyNetworkInterfaceRequestOnVMTemplateSpec` and only after everything finished to update the original `vm`.